### PR TITLE
Add Nextcloud Talk support

### PR DESCRIPTION
**Pull request by gary-kim**
~~This is still a work in progress but I welcome any feedback anyone may have :).~~

### Ready for review :)

~~I forgot about documentation, gimme a sec~~

This adds basic support for Nextcloud Talk. I'm hoping to add support for stuff like replies and such in a future PR.

Signed-off-by: Gary Kim <gary@garykim.dev>

### CODE REVIEW

File: bridge/talk/talk.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			if msg.MessageType != ocs.MessageComment || msg.ActorID == b.user.User {
+				continue
+			}
+			remoteMessage := config.Message{
+				Text:      msg.Message,
+				Channel:   newRoom.room.Token,
+				Username:  msg.ActorDisplayName,
+				UserID:    msg.ActorID,
+				Account:   b.Account,
+				Timestamp: time.Unix(int64(msg.Timestamp), 0),
```
Comment (42wim) [inline bridge/talk/talk.go:76]: You don't need to set the Timestamp here, the gateway does this

**Comment by 42wim** [comment]
Cool!
The WIP looks good, added a small comment.
The linter is also complaining so please take a look at that too (we don't use named returns in our codebase)

File: bridge/talk/talk.go (hunk 1 above)
Comment (gary-kim) [reply bridge/talk/talk.go:76]: Okay, thank you!

**Comment by gary-kim** [comment]
Tested and working. I'll write a README for the library in the next 24 hours and tag it then this will be complete!

File: bridge/talk/talk.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+}
+
+func (b *Btalk) Connect() error {
+	b.Log.Info("Connecting")
+	b.user = talk.NewUser(b.GetString("Server"), b.GetString("Login"), b.GetString("Password"))
+	_, err := b.user.Capabilities()
+	if err != nil {
+		b.Log.Info("Connected")
```
Comment (42wim) [inline bridge/talk/talk.go:37]: shouldn't this be `return err` ?
The err isn't nil so it's actually an error.

File: bridge/talk/talk.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	} else {
+		b.Log.Info("Cannot connect")
```
Comment (42wim) [inline bridge/talk/talk.go:39]: This should also change then

File: bridge/talk/talk.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
+package talk
```
Comment (42wim) [inline bridge/talk/talk.go:1]: As talk is too generic, I think we should better rename this bridge and package to `nctalk`

**Comment by 42wim** [comment]
Thank you! some small issues and it can get merged :)

**Comment by eneiluj** [comment]
Hey there. I just tried it between NC Talk and Mattermost and it works great!

I guess you already know all that and want to take care of issues later. Just listing what I see:

* Nick message prefix works both ways.
* Emojis work both ways.
* When mentioning someone in the Talk room, it appears like that in the Mattermost channel: `{mention-user1} blabla`. Could it be replaced with something like `@username blabla`?
* When joining a file in the Talk room (new file or from storage), it shows `{file}`. Maybe those could be ignored. What do you think?
* Joining a file in the Mattermost channel displays an empty message in the Talk room. I guess it's a choice made in Mattermost backend.


Pretty exciting stuff. Cheers!

File: bridge/talk/talk.go (hunk 2 above)
Comment (gary-kim) [reply bridge/talk/talk.go:37]: Whoops! I have no idea what I was thinking here. Will fix

**Comment by gary-kim** [comment]
Thank you for your far more comprehensive tests @eneiluj 

I honestly forgot to test how files will send. Let me try that too.

**Pull request merged**